src/averageAges.js

```diff
@@ -55,10 +79,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const withMothers = people.filter(child =>
+    !onlyWithSon
+      ? people.some((person) => child.mother === person.name)
+      : people.some((person) => (child.mother === person.name)
+        && child.sex === 'm')
```

Comment:
* Consider using constants with meaningful names instead of magic strings.
* use here isMan or isWoman, avoid repeating code